### ISSUE TITLE
How to re-layout the graph after deleting some nodes ?

### ISSUE THREAD

**Issue by Nazeh-Taha** [wbkd/react-flow#1313]
### Discussed in https://github.com/wbkd/react-flow/discussions/1312

<div type='discussions-op-text'>

<sup>Originally posted by **Nazeh-Taha** June 27, 2021</sup>
How to re-layout the graph after deleting some nodes?
```
const getLayoutedElements = (els, direction = "BT") => {
    dagreGraph.setGraph({ rankdir: direction });
    els.forEach((el) => {
      if (el.type !== "strategynode") {
        if (isNode(el)) {
          dagreGraph.setNode(el.id, {
            width: findWidth(el),
            height: 130,
          });
        } else {
          dagreGraph.setEdge(el.source, el.target);
        }
      }
    })

    dagre.layout(dagreGraph);

    return els.map((el) => {
      if (el.type !== "strategynode") {
        if (isNode(el)) {
          const nodeWithPosition = dagreGraph.node(el.id);
          el.position = {
            x: nodeWithPosition.x - findWidth(el) / 2 + Math.random() / 1000,
            y: nodeWithPosition.y - nodeHeight / 2,
          };
        }
      }
      return el;
    });
  };
```

` const onElementsRemove = (elementsToRemove) => {
    setElements((els) => removeElements(elementsToRemove, els));
    getLayoutedElements(elements);
  };`

Everything is good:
![1](https://user-images.githubusercontent.com/54739625/123546432-25c0e780-d765-11eb-9ec6-3af933604001.jpg)

After delete nodes:
![2](https://user-images.githubusercontent.com/54739625/123546454-3a9d7b00-d765-11eb-9596-196f055b43f7.jpg)

After adding the same nodes again:
![3](https://user-images.githubusercontent.com/54739625/123546472-53a62c00-d765-11eb-8019-aefd8e49ccdf.jpg)
</div>

**Comment by moklick** [comment]
closed in favour of #1312

**Comment by AlexanderPershin** [comment]
@Nazeh-Taha I solved it by moving entire `dagreGraph` initialization into `getLayoutedElements` function, like so:
```
export const getLayoutedElements = (
    nodes: Node[],
    edges: Edge[],
    direction = 'TB'
) => {
    const dagreGraph = new dagre.graphlib.Graph();
    dagreGraph.setDefaultEdgeLabel(() => ({}));
    const isHorizontal = direction === 'LR';

    dagreGraph.setGraph({ rankdir: direction });
...
```
